temporalio/worker/replayer.py

```diff
@@ -94,36 +77,82 @@ def config(self) -> ReplayerConfig:
+        *,
+        raise_on_replay_failure: bool = True,
+    ) -> WorkflowReplayResult:
         """Replay a workflow for the given history.
 
         Args:
             history: The history to replay. Can be fetched directly, or use
-              :py:meth:`WorkflowHistory.from_json` to parse a history downloaded via `tctl` or the
-              web ui.
+                :py:meth:`temporalio.client.WorkflowHistory.from_json` to parse
+                a history downloaded via ``tctl`` or the web UI.
+            raise_on_replay_failure: If ``True`` (the default), this will raise
+                a :py:attr:`WorkflowReplayResult.replay_failure` if it is
+                present.
         """
 
-        async def gen_hist():
+        async def history_iterator():
             yield history
 
-        await self.replay_workflows(gen_hist())
+        async with self.workflow_replay_iterator(history_iterator()) as replay_iterator:
+            async for result in replay_iterator:
+                if raise_on_replay_failure and result.replay_failure:
+                    raise result.replay_failure
+                return result
+            # Should never be reached
+            raise RuntimeError("No histories")
 
     async def replay_workflows(
-        self, histories: AsyncIterator[WorkflowHistory]
+        self,
+        histories: AsyncIterator[temporalio.client.WorkflowHistory],
+        *,
+        raise_on_replay_failure: bool = True,
     ) -> WorkflowReplayResults:
-        """Replay a workflow for the given history.
+        """Replay workflows for the given histories.
+
+        This is a shortcut for :py:meth:`workflow_replay_iterator` that iterates
+        all results and aggregates information about them.
 
         Args:
             histories: The histories to replay, from an async iterator.
+            raise_on_replay_failure: If ``True`` (the default), this will raise
+                the first replay failure seen.
+
+        Returns:
+            Aggregated results.
         """
-        current_run_results = WorkflowReplayResults(dict())
-        # Create bridge worker and workflow worker
-        tq = f"replay-{self._config['build_id']}"
-        (bridge_worker, pusher,) = temporalio.bridge.worker.Worker.for_replay(
+        async with self.workflow_replay_iterator(histories) as replay_iterator:
+            replay_failures: Dict[str, Exception] = {}
+            async for result in replay_iterator:
+                if result.replay_failure:
+                    if raise_on_replay_failure:
+                        raise result.replay_failure
+                    replay_failures[result.history.run_id] = result.replay_failure
+            return WorkflowReplayResults(replay_failures=replay_failures)
+
+    @asynccontextmanager
+    async def workflow_replay_iterator(
+        self, histories: AsyncIterator[temporalio.client.WorkflowHistory]
+    ) -> AsyncIterator[AsyncIterator[WorkflowReplayResult]]:
```

Comment:
Double AsyncIterator?

Reply:
Really a context manager of an iterator. So it's `async with workflow_replay_iterator(histories) as iterator:`. But context manager helper is represented as a generator/iterator of a single yielded item.